### PR TITLE
Add index_select shape expression

### PR DESCRIPTION
**Pull request by ftxj**
Since the output shape of `index_select` operator is a combination of input and index, we add this shape expression.

### CODE REVIEW

File: third_party/nvfuser/csrc/graph_fuser.cpp [hunk 1]
```diff
@@ -1084,7 +1084,24 @@ struct CudaGraphFuser {
         shape_of.emplace(n->output(), size);
         continue;
       }
+      if (n->kind() == aten::index_select) {
+        TORCH_INTERNAL_ASSERT(
+            shape_of.count(n->input(0)) > 0,
+            "buildShapeExpressions failed at accessing input shapes");
+        Node* dim_const = graph->createClone(n->input(1)->node(), map_inputs);
```
Comment (jjsjann123) [inline third_party/nvfuser/csrc/graph_fuser.cpp:1096]: nitpick, can we add a check `n->input(1)->node()` is a constant node?

**Review by jjsjann123 — COMMENTED** [review]
Looks pretty straightforward. Can we add a test for this?

If you have a small~ish example, we can add that in `test_torchscript.py`.

**Comment by ftxj** [comment]
> 

Sorry for not giving a test, the commit has been updated

File: third_party/nvfuser/csrc/graph_fuser.cpp (hunk 1 above)
Comment (ftxj) [reply third_party/nvfuser/csrc/graph_fuser.cpp:1096]: Done

File: third_party/nvfuser/python_tests/test_torchscript.py [hunk 2]
```diff
@@ -4201,6 +4201,25 @@ def t(x: torch.Tensor, y: torch.Tensor, ind: torch.Tensor):
...
+        ind = torch.randint(0, 68, (130,), device="cuda").to(dtype=torch.int)
+
+        def t(x: torch.Tensor, y: torch.Tensor, ind: torch.Tensor):
+            o = torch.mul(x, y)
+            o = torch.index_select(o, 0, ind)
+            return o.size()
+
+        t_jit = torch.jit.script(t)
+        jit_o = t_jit(x, y, ind)
+        jit_o = t_jit(x, y, ind)
```
Comment (jjsjann123) [inline third_party/nvfuser/python_tests/test_torchscript.py:4219]: Looks good. Can we also assert that we actually have the the short-cut for shape in-place?
something like this vvv.
https://github.com/csarofeen/pytorch/blob/75ce4aeb896c7bc0558e677f2b548f0312f791ae/third_party/nvfuser/python_tests/test_torchscript.py#L3929-L3930

**Review by jjsjann123 — APPROVED** [review]
LGTM. Stamping.

One nitpick on python test~

File: third_party/nvfuser/python_tests/test_torchscript.py (hunk 2 above)
Comment (ftxj) [reply third_party/nvfuser/python_tests/test_torchscript.py:4219]: Thanks for your tips! Done.

**Comment by jjsjann123** [comment]
@ftxj Feel free to merge it once tests passed locally~~

**Comment by ftxj** [comment]
> @ftxj Feel free to merge it once tests passed locally~~

Sorry, I don't have the write permission...

**Comment by jjsjann123** [comment]
Looks like Naoya beats me to it~

**Pull request merged**